### PR TITLE
Upgrade GroovyCompileOptions

### PR DESCRIPTION
**Pull request by asodja**

### CODE REVIEW

**Comment by asodja** [comment]
@bot-gradle test this

**Comment by asodja** [comment]
@bot-gradle test QF

**Comment by asodja** [comment]
@bot-gradle test this

**Comment by asodja** [comment]
@bot-gradle test this

File: platforms/jvm/language-groovy/src/main/java/org/gradle/api/tasks/compile/GroovyCompileOptions.java [hunk 1]
```diff
@@ -81,80 +70,58 @@ protected ObjectFactory getObjectFactory() {
...
+    public abstract Property<Boolean> getListFiles();
 
-    /**
-     * Sets whether to print which source files are to be compiled. Defaults to {@code false}.
-     */
-    public void setListFiles(boolean listFiles) {
-        this.listFiles = listFiles;
+    @Internal
+    public Property<Boolean> getIsListFiles() {
+        return getListFiles();
```
Comment (lptr) [inline platforms/jvm/language-groovy/src/main/java/org/gradle/api/tasks/compile/GroovyCompileOptions.java:104]: ❓ Shouldn't this be deprecated?

**Comment by lptr** [comment]
@bot-gradle test QFL please.

**Comment by asodja** [comment]
@bot-gradle test this

**Comment by asodja** [comment]
@bot-gradle test this

**Comment by asodja** [comment]
@bot-gradle test this

**Pull request merged**